java/ql/src/experimental/Security/CWE/CWE-522/InsecureBasicAuth.ql

```diff
@@ -66,25 +72,17 @@ class URLConstructor extends ClassInstanceExpr {
  * Gets a regular expression for matching private hosts.
  */
 private string getPrivateHostRegex() {
-  result = "localhost(/.*)?" or
-  result = "127\\.0\\.0\\.1(/.*)?" or // IPv4 patterns
-  result = "10(\\.[0-9]+){3}(/.*)?" or
-  result = "172\\.16(\\.[0-9]+){2}(/.*)?" or
-  result = "192.168(\\.[0-9]+){2}(/.*)?" or
-  result = "\\[0:0:0:0:0:0:0:1\\](/.*)?" or // IPv6 patterns
-  result = "\\[::1\\](/.*)?"
+  result = "(?i)localhost([:/].*)?|127\\.0\\.0\\.1([:/].*)?|10(\\.[0-9]+){3}([:/].*)?|172\\.16(\\.[0-9]+){2}([:/].*)?|192.168(\\.[0-9]+){2}([:/].*)?|\\[0:0:0:0:0:0:0:1\\]([:/].*)?|\\[::1\\]([:/].*)?" 
```

Comment:
Also this regex currently uses capturing groups. Using non-capturing groups might slightly increase performance:
```suggestion
  result = "(?i)localhost(?:[:/].*)?|127\\.0\\.0\\.1(?:[:/].*)?|10(?:\\.[0-9]+){3}(?:[:/].*)?|172\\.16(?:\\.[0-9]+){2}(?:[:/].*)?|192.168(?:\\.[0-9]+){2}(?:[:/].*)?|\\[0:0:0:0:0:0:0:1\\](?:[:/].*)?|\\[::1\\](?:[:/].*)?" 
```

Reply:
I've added non-capturing groups as per the suggestion.